tombstones.go

```diff
@@ -150,36 +152,50 @@ func readTombstones(dir string) (memTombstones, error) {
 			return nil, d.err()
 		}
 
-		stonesMap.add(k, Interval{mint, maxt})
+		stonesMap.addInterval(k, Interval{mint, maxt})
 	}
 
 	return stonesMap, nil
 }
 
-type memTombstones map[uint64]Intervals
-
-var emptyTombstoneReader = memTombstones{}
+type memTombstones struct {
+	mts map[uint64]Intervals
+	mtx sync.RWMutex
+}
 
-// EmptyTombstoneReader returns a TombstoneReader that is always empty.
-func EmptyTombstoneReader() TombstoneReader {
-	return emptyTombstoneReader
```

Comment:
We should not be removing exported functionality without replacement.

Reply:
@codwu in that case we can export newMemTombstones -> NewMemTombstones

Reply:
@krasi-georgiev  ok, I made a new commit.

Reply:
@fabxc do you remember why was this exposed and if not why should we keep it exposed?
I guess if anyone is actually using it we can always re-expose it given it is really needed.